### PR TITLE
Add video to homepage hero

### PR DESCRIPTION
**Pull request by karlhorky**
Hey @Newbie012 👋 Just a quick PR adding the video from [your tweet](https://twitter.com/CoEliya/status/1564036318229073921) (not working yet!)

https://user-images.githubusercontent.com/1935696/190372679-b02a1e43-ec6f-427d-ac0f-1586aa5e5c5a.mp4

---

Ah, I guess VitePress's default theme's `home` layout doesn't support HTML in there (see screenshot below): https://github.com/vuejs/vitepress/issues/1340

Hm, I'm not entirely sure what to do here. Happy to follow some suggestions!

**Alternative:** Switch out the large logo for a GIF instead... 🤔 VitePress would support this

Screen Shot 2022-09-14 at 18 19 47

### CODE REVIEW

**Comment by karlhorky** [comment]
@Newbie012 any opinion on this here?

In https://github.com/vuejs/vitepress/issues/1340 they say that they would like to support a video, but who knows when this feature will appear.

Maybe we could still add a video (screen recording) on the homepage in the meantime? I think it would show quickly and simply what SafeQL does. Even just using your original video from the tweet would be a great start!

**Comment by Newbie012** [comment]
I like the idea, although I think we should show a simpler and shorter videos with higher quality (the one in Twitter has a quite poor quality) that still emphasizes the power of this tool.

There's this tweet as well - https://twitter.com/CoEliya/status/1570051268294463490?s=20&t=zn3SAQUNODabhAgat7aN1A

**Comment by karlhorky** [comment]
Ok good, sounds fine to me.

Do you know VitePress well? (eg. creating a custom theme / layout based on the `home` layout)

Do you have a suggestion for how we can add a video to the homepage?

**Comment by Newbie012** [comment]
We have a few options:

1. Wait for vitepress for official support.
2. Copy and edit https://github.com/vuejs/vitepress/blob/main/src/client/theme-default/components/VPHero.vue and then use it instead.
3. Add the video to the bottom of the page.

Since I'm not in a hurry, then I wouldn't mind going with option one.

I think I have the original video with a higher quality.
I'll have to check.

**Comment by karlhorky** [comment]
Seems like VitePress supports this now!

- https://github.com/vuejs/vitepress/pull/1528
- https://github.com/vuejs/vitepress/issues/1062

Although I'm not super familiar with VitePress and extending slots - not sure exactly how to implement this. I've asked in https://github.com/vuejs/vitepress/issues/1340#issuecomment-1366698070

**Comment by karlhorky** [comment]
@Newbie012  Added a `Layout.vue` according to the recommendation by @brc-dd over here:

- https://github.com/vuejs/vitepress/issues/1340#issuecomment-1463522111

Maybe ready for review now?

**Comment by karlhorky** [comment]
Oh, [the deploy preview](https://safeql-kxpskh2ay-safeql.vercel.app/) for that does not show the video, did I do something wrong?

Screenshot 2023-03-10 at 10 48 57

I thought it would take place of the image on the right. Or maybe VitePress needs to be updated? 🤔

**Comment by brc-dd** [comment]
@karlhorky Update lock files too.

**Comment by karlhorky** [comment]
Ok, CI is passing now, not sure what needs to be done to trigger another deploy...

**Comment by brc-dd** [comment]
A recommendation: Split tagline (docs/index.md) to multiple lines (currently it's overlapping with the video):

```yaml
  tagline: |
    An ESLint plugin for writing SQL queries
    in a type-safe way.
```

File: docs/.vitepress/theme/Layout.vue [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import DefaultTheme from "vitepress/theme";
+</script>
+
+<template>
+  <DefaultTheme.Layout>
+    <template #home-hero-image>
+      <video src="/video.mp4" autoplay loop muted />
+    </template>
+  </DefaultTheme.Layout>
+</template>
```
Comment (brc-dd) [inline docs/.vitepress/theme/Layout.vue:11]:
```suggestion
<script setup>
import DefaultTheme from "vitepress/theme";
</script>

<template>
  <DefaultTheme.Layout>
    <template #home-hero-image>
      <video autoplay loop muted src="/video.mp4" class="video" />
    </template>
  </DefaultTheme.Layout>
</template>

<style scoped>
.video {
  position: absolute;
  top: 50%;
  left: 50%;
  max-height: 192px;
  transform: translate(-50%, -50%);
}

@media (min-width: 640px) {
  .video {
    max-height: unset;
  }
}
</style>
```

**Comment by karlhorky** [comment]
Looks like it's working, thanks!! Thanks for the help and improvements @brc-dd ! 🙌

https://user-images.githubusercontent.com/1935696/224289567-de011e5e-3f78-48af-b75f-47452ddef485.mp4

**Comment by karlhorky** [comment]
No worries, thanks for getting back to it!

**Comment by Newbie012** [comment]
Thanks everyone , it looks great 😃

**Comment by karlhorky** [comment]
Glad to help, thanks for the review and merge!

**Pull request merged**